tests/host_tools/fillmem.c

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+
+
+#define MB (1024 * 1024)
+
+
+int fill_mem(int mb_count) {
+    int i, j;
+    char *ptr = NULL;
+    for(j = 0; j < mb_count; j++) {
+        ptr = NULL;
+        while (ptr == NULL) {
+            // We can't map the whole chunk of memory at once because
+            // on aarch64 when we try to allocate a large amount of
+            // memory to OOM the system, it often results in the
+            // process getting killed by a SIGSEGV instead of the
```

Comment:
Interesting, any idea why?

Reply:
I appreciate the comments here, but it is a bit vague (often results?) and does not highlight the root cause.

Reply:
It works as expected on x86_64, but on arm64 there is an almost arbitrary interval of values for which allocations will succeed but will result in an OOM death of the process, but if we try to allocate memory of sizes above that interval it results in a straight SIGSEGV. From my experiments, the window for this successful death by OOM killer seems to be from the point of memory pressure to about 18 MB above that, and of course the point at which the system enters a memory pressure state varies from run to run. I don't know why this happens and I don't think it's relevant for this PR.

Reply:
I found out that the SIGSEGV mentioned in this comment came from the fact that I was wrong to check the `mmap` result like this (`ptr == NULL`). What I should have done was `ptr == MAP_FAILED`.

The root cause of the issue is that when the program is used with the purpose of making the system enter an OOM state, it is trying to allocate and demand-page in more memory than is available. If it does this as one big allocation, it will fail since the memory manager will just refuse the allocation with `ENOMEM`. In this case it makes sense to try multiple small allocations since they are much more likely to succeed and this is by no means a performance critical code path.

Reply:
Ok, that makes sense! Please update the comment to reflect this as well. 👍